virttest/qemu_storage.py

```diff
@@ -25,22 +25,96 @@
+        secret_type = 'password'
+        return storage.CephAccessInfo.access_info_define_by_params(image,
+                                                                   params,
+                                                                   secret_type)
+    return None
+
+
+def filename_to_file_opts(filename):
+    """Convert filename into file opts, used by both qemu-img and qemu-kvm"""
+    file_opts = {}
+    if not filename:
+        file_opts = {}
+    elif filename.startswith('iscsi:'):
+        filename_pattern = re.compile(
+            r'iscsi://(?P<auth_portal>.+?)/(?P<target>.+?)/(?P<lun>\d+)')
+        matches = filename_pattern.match(filename)
+        if matches:
+            file_opts = {'transport': 'tcp',
+                         'target': matches.group('target'),
+                         'lun': matches.group('lun')}
+
+            auth_portal = matches.group('auth_portal')
+            auth_portal_pattern = re.compile(
+                r'(?P<user>.+?):(?P<password>.+?)@(?P<portal>.+)')
+            auth_portal_matches = auth_portal_pattern.match(auth_portal)
+            if auth_portal_matches:
+                file_opts['user'] = auth_portal_matches.group('user')
+                file_opts['portal'] = auth_portal_matches.group('portal')
+            else:
+                file_opts['portal'] = auth_portal
+    elif filename.startswith('rbd:'):
+        filename_pattern = re.compile(r'rbd:(?P<pool>.+)/(?P<image>.+)')
+        if re.search(r':conf=', filename):
+            filename_pattern = re.compile(
+                r'rbd:(?P<pool>.+?)/(?P<image>.+):conf=(?P<conf>.+)')
+        matches = filename_pattern.match(filename)
+        file_opts = matches.groupdict() if matches else {}
+    else:
+        file_opts['filename'] = filename
+
+    return file_opts
+
 
 def _get_image_meta(image, params, root_dir):
     """Retrieve image meta dict."""
-    image_filename = storage.get_image_filename(params, root_dir)
-    image_format = params.get("image_format", "qcow2")
-    image_encryption = params.get("image_encryption", "off")
     meta = collections.OrderedDict()
-    secret = storage.ImageSecret.image_secret_define_by_params(image, params)
-    if image_format == "qcow2" and image_encryption == "luks":
-        meta["encrypt.key-secret"] = secret.aid
-    meta["driver"] = image_format
     meta["file"] = collections.OrderedDict()
-    meta["file"]["driver"] = "file"
-    meta["file"]["filename"] = image_filename
+
+    filename = storage.get_image_filename(params, root_dir)
+    file_opts = filename_to_file_opts(filename)
+
+    meta_file = {}
+    meta_file.update(file_opts)
+
+    enable_ceph = params.get("enable_ceph", "no") == "yes"
+    enable_iscsi = params.get("enable_iscsi", "no") == "yes"
+
+    if enable_ceph:
+        meta_file['driver'] = 'rbd'
+    elif enable_iscsi:
+        meta_file['driver'] = 'iscsi'
+    else:
+        meta_file['driver'] = 'file'
```

Comment:
You can let `filename_to_file_opts()` set this thing.

Reply:
filename_to_file_opts is used by both qemu_storage and qcontainer, the former needs the 'driver' while the latter doesn't need it(driver is set by TYPE). so I remove 'driver' from the function. 

But it's OK to include the 'driver' in filename_to_file_opts, QBlockdevNode in qcontainer will set 'driver' twice with the same value.